paymentapi/src/main/java/fi/hel/verkkokauppa/payment/paytrail/converter/impl/PaytrailCreatePaymentPayloadConverter.java

```diff
@@ -70,7 +70,10 @@ private ArrayList<PaymentItem> createPaytrailPaymentItemsFromOrderItems(Paytrail
             paymentItem.setVatPercentage(Integer.valueOf(orderItemDto.getVatPercentage()));
             paymentItem.setProductCode(orderItemDto.getProductId());
             paymentItem.setDescription(orderItemDto.getProductName());
-            paymentItem.setOrderId(orderItemDto.getOrderId());
+
+            if (context.isUseShopInShop()) {
```

Comment:
I think paymentItem needs orderId allways, so no need to make some kind of if staments about it?

Reply:
Add new field to paymentItem like canUseShopInShop boolean etc. then you can just add return value of that context funtion as field value